multicast/recv.py

```
@@ -211,7 +212,95 @@
+_w_unspec_bind: str = "\n".join([
+	"Lazy call to multicast.joinstep with unspecified bind group.",
+	f"Will bind to {multicast._MCAST_DEFAULT_BIND_IP}.",  # skipcq: PYL-W0212 - module ok
+	"Tip: Pass a value for bind_group to suppress this message",
+	"(such as 'bind_group=multicast._MCAST_DEFAULT_BIND_IP')",
+])
+
+
+_w_non_multicast = f"{_w_prefix}\nJust use socket.Socket.bind(...) for non-multicast networking."
+
+
+def _validate_join_args(groups=None, port=None, iface=None, bind_group=None, isock=None) -> tuple:
```

Ruff (0.8.2)

Missing type annotation for function argument `groups`

(ANN001)